### PR TITLE
Some cleanup

### PR DESCRIPTION
**Pull request by ArthurZucker**
IMO it makes more sense that STR just implements `__str__` while `AutoDisplay` implements display, and AutoDebug debug. 

This allows native rust objects to benefit from AutoDisplay. 

Also fixed a bug where `#[default]` for example would raise an error when parsing meta args, some refactoring here and there.

### CODE REVIEW

File: pyo3_special_method_derive_example/src/lib.rs [hunk 1]
```diff
@@ -1,55 +1,71 @@
...
+    #[default]
+    London
+}
+
+#[derive(Default, AutoDebug, AutoDisplay)]
+pub struct Country {
+    capital : Option<City>,   
+    pub regions: Vec<City> ,
+    pub(crate) hash: u128,
+    inhabitants: HashMap<City, u32>, // it's private but I still want to be able to display it!
```
Comment (ArthurZucker) [inline pyo3_special_method_derive_example/src/lib.rs:18]: TODO, `#[auto_display]` on top of variant / field to force displaying

File: pyo3_special_method_derive_example/src/lib.rs [hunk 2]
```diff
@@ -1,55 +1,71 @@
...
+    pub(crate) hash: u128,
+    inhabitants: HashMap<City, u32>, // it's private but I still want to be able to display it! 
+}
+
+#[derive(AutoDisplay, AutoDebug)]
+pub struct Region;
+
+// I want the display to be "MyObject." not MyObjectWrapper
+#[derive(AutoDisplay, AutoDebug)]
+pub enum MyObjectWrapper {
```
Comment (ArthurZucker) [inline pyo3_special_method_derive_example/src/lib.rs:26]: something like 
```suggestion
#[auto_display("MyObject")]
pub enum MyObjectWrapper {
```

Comment (EricLBuehler) [reply pyo3_special_method_derive_example/src/lib.rs:26]: That would be great!

File: pyo3_special_method_derive_example/src/lib.rs (hunk 1 above)
Comment (EricLBuehler) [reply pyo3_special_method_derive_example/src/lib.rs:18]: That makes sense, perhaps even something like `#[pyo3_smd(force_include)]` which could be used by other derive macros would be helpful? But maybe we want this level of granularity.

Comment (haixuanTao) [reply pyo3_special_method_derive_example/src/lib.rs:18]: +1 on this!

**Pull request closed without merging**